core/modules/query/QueryTemplate.cc

```diff
@@ -48,43 +48,56 @@
 #include "query/ColumnRef.h"
 #include "query/TableRef.h"
 
-
 namespace lsst {
 namespace qserv {
 namespace query {
 
+namespace {
+
+LOG_LOGGER getLogger() {
+    static LOG_LOGGER logger = LOG_GET("lsst.qserv.query.QueryTemplate");
+    return logger;
+}
+
+
 struct SpacedOutput {
+
     SpacedOutput(std::ostream& os_, std::string sep_=" ")
-        : os(os_), sep(sep_), count(0) {}
-    void operator()(std::string const& s) {
-        if(s.empty()) return;
+        : os(os_), sep(sep_) {}
+
+    void operator()(std::shared_ptr<QueryTemplate::Entry> entry) {
+        if(!entry) { throw std::invalid_argument("NULL QueryTemplate::Entry"); }
+        //if(e->isDynamic()) { os << "(" << count << ")"; }
 
-        if(!last.empty() && sql::sqlShouldSeparate(last, *(last.end()-1), s[0]))  {
+        std::string const& entry_str = entry->getValue();
+        LOGF(getLogger(), LOG_LVL_TRACE, "entry: %1%" % entry_str);
+        if(entry_str.empty()) return;
+
+        if(!last_entry.empty() && sql::sqlShouldSeparate(last_entry, last_entry.back(), entry_str.at(0)))  {
             os << sep;
         }
-        os << s;
-        last = s;
-    }
-    void operator()(std::shared_ptr<QueryTemplate::Entry> e) {
-        if(!e) { throw std::invalid_argument("NULL QueryTemplate::Entry"); }
-        //if(e->isDynamic()) { os << "(" << count << ")"; }
-        (*this)(e->getValue());
-        ++count;
+        os << entry_str;
+        last_entry = entry_str;
     }
 
     std::ostream& os;
-    std::string last;
+    std::string last_entry;
     std::string sep;
-    int count;
 };
 
-template <typename C>
-std::string outputString(C& c) {
+
+std::string stringify(QueryTemplate::EntryPtrVector const& v) {
```

Comment:
renamed it to toString(), I wanted toString() to be a method and stringify() a function